src/encoder.rs

```diff
@@ -2303,44 +2309,63 @@ fn encode_partition_bottomup(seq: &Sequence, fi: &FrameInvariants, fs: &mut Fram
         let mode_decision = rdo_mode_decision(seq, fi, fs, cw, bsize, bo, spmvs, false).part_modes[0].clone();
 
         rd_cost = mode_decision.rd_cost + cost;
+        best_rd = rd_cost;
 
         encode_block_with_modes(seq, fi, fs, cw, w_pre_cdef, w_post_cdef, bsize, bo,
                             &mode_decision);
+
         best_decision = mode_decision;
+        best_pred_modes.push(best_decision.clone());
     }
 
     // Code a split partition and compare RD costs
     if can_split {
-        cw.rollback(&cw_checkpoint);
-        w_pre_cdef.rollback(&w_pre_checkpoint);
-        w_post_cdef.rollback(&w_post_checkpoint);
-
-        let nosplit_rd_cost = rd_cost;
-        partition = PartitionType::PARTITION_SPLIT;
-        {
-        subsize = bsize.subsize(partition);
-        let hbsw = subsize.width_mi(); // Half the block size width in blocks
-        let hbsh = subsize.height_mi(); // Half the block size height in blocks
+        for &partition in RAV1E_PARTITION_TYPES {
+            if partition == PartitionType::PARTITION_NONE {
+                continue;
+            }
+            assert!(bsw == bsh);
```

Comment:
Please add some spacing around inner scopes in this area, for ease of reading.

Reply:
sure-.